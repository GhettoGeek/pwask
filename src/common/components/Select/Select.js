import React from 'react'
import PropTypes from 'prop-types'
import { Select } from '@material-ui/core'

function EnhancedSelect({
  id, items, ...props
}) {
  const [value, setValue] = React.useState('')
  const handleChange = (event) => setValue(event.target.value)

  return (
    <Select
      native
      id={id}
      variant="outlined"
      value={value}
      onChange={handleChange}
      {...props}
    >
      {items.map((item) => (
        <option key={item.value} value={item.value}>
          {item.label}
        </option>
      ))}
    </Select>
  )
}

EnhancedSelect.propTypes = {
  id: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    value: PropTypes.string,
  })),
}

EnhancedSelect.defaultProps = {
  id: 'standard-select-native',
  items: [],
}

export default EnhancedSelect
