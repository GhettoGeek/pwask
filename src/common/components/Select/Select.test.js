import React from 'react'
import { shallow } from 'enzyme'
import Select from './Select'

describe('Select', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(<Select />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
