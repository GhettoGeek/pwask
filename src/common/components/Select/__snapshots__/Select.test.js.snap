// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly the component 1`] = `"<div class=\\"MuiInputBase-root MuiOutlinedInput-root\\"><select class=\\"MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect\\"></select><svg class=\\"MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg><fieldset aria-hidden=\\"true\\" style=\\"padding-left:8px\\" class=\\"PrivateNotchedOutline-root-55 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legend-56\\" style=\\"width:0.01px\\"><span>&#8203;</span></legend></fieldset></div>"`;
